### PR TITLE
Add mistral large sut hosted on vertex ai

### PR DESCRIPTION
**Pull request by rogthefrog**
[Story](https://github.com/mlcommons/modelbench/issues/607)

### CODE REVIEW

File: plugins/mistralai_sut/modelgauge/suts/vertexai_client.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        return headers
+
+    def request(self, req: dict) -> dict:
+        try:
+            client = httpx.Client()
+            response = client.post(self._build_endpoint_url(), json=req, headers=self._headers(), timeout=None)
+            if response.status_code == 200:
+                return response.json()
+            else:
+                raise RuntimeError(f"VertexAI response code {response.status_code}")
```
Comment (wpietri) [inline plugins/mistralai_sut/modelgauge/suts/vertexai_client.py:81]: Any production SUT will need to have some retry logic sufficient to overcome transient errors and to keep retrying on rate limits.

**Review by wpietri — CHANGES_REQUESTED** [review]

File: plugins/mistralai_sut/modelgauge/suts/vertexai_client.py (hunk 1 above)
Comment (rogthefrog) [reply plugins/mistralai_sut/modelgauge/suts/vertexai_client.py:81]: OK

File: plugins/mistralai_sut/modelgauge/suts/vertexai_client.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+class VertexAIClient:
+    def __init__(
+        self,
+        publisher: str,
+        model_name: str,
+        model_version: str,
+        streaming: bool,
+        api_key: VertexAIAPIKey,
+        project_id: VertexAIProjectId,
+        region: str = os.environ.get("GOOGLE_REGION"),
```
[the remark this answers is not in the document]
Comment (rogthefrog) [reply plugins/mistralai_sut/modelgauge/suts/vertexai_client.py:43]: OK

File: plugins/mistralai_sut/modelgauge/suts/vertexai_client.py (hunk 1 above)
Comment (rogthefrog) [reply plugins/mistralai_sut/modelgauge/suts/vertexai_client.py]: I've added retry logic to the MistralAI sut, and left a comment to revisit this client if/when we use VertexAI again in the future.

**Comment by rogthefrog** [comment]
@wpietri @bkorycki could you please take another look, and approve if it's OK (to remove "merging is blocked")?

**Comment by wpietri** [comment]
> could you please take another look, and approve if it's OK (to remove "merging is blocked")?

I have approved, so I think it's down to @bkorycki  now.

File: plugins/vertexai/modelgauge/suts/vertexai_client.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        self.streaming = streaming
+        if isinstance(region, str):
+            self.region = region
+        elif isinstance(region, VertexAIRegion):
+            self.region = region.value
+        else:
+            raise ValueError("Incorrect GCP region.")
+
+    def _get_access_token(self) -> str:
+        credentials, _ = google.auth.default(scopes=["https://www.googleapis.com/auth/cloud-platform"])
```
Comment (bkorycki) [inline plugins/vertexai/modelgauge/suts/vertexai_client.py:66]: How does this work? Does it require you to have credentials setup in gcloud CLI?

File: plugins/mistral/modelgauge/suts/mistral_sut.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+model_name = "ministral-8b"
+model_version = "2410"
+model_uid = f"{model_name}-latest"
+SUTS.register(MistralAISut, model_uid, model_uid, model_version, MISTRAL_API_KEY)
+
+
+model_name = "mistral-large"
+model_version = "2411"
+model_uid = f"{model_name}-latest"
```
Comment (bkorycki) [inline plugins/mistral/modelgauge/suts/mistral_sut.py:86]: The version should be explicit in the uid e.g. `mistral-large-2411`

File: plugins/vertexai/modelgauge/suts/vertexai_mistral_sut.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        return SUTResponse(completions=completions)
+
+
+VERTEX_KEY = InjectSecret(VertexAIAPIKey)
+VERTEX_PROJECT_ID = InjectSecret(VertexAIProjectId)
+VERTEX_REGION = InjectSecret(VertexAIRegion)
+
+model_name = "mistral-large"
+model_version = "2407"
+model_uid = f"{model_name}-{model_version}"
```
Comment (bkorycki) [inline plugins/vertexai/modelgauge/suts/vertexai_mistral_sut.py:108]: I think @bollacker has mentioned in the past that the provider should also be a part of the UID. Especially if we supply multiple providers for one SUT.

**Review by bkorycki — CHANGES_REQUESTED** [review]

**Comment by bkorycki** [comment]
I get an error when I run `poetry run modelgauge run-sut --sut mistral-large-latest`:
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for TextPrompt
text
  Input should be a valid string [type=string_type, input_value=None, input_type=NoneType]
```

File: plugins/vertexai/modelgauge/suts/vertexai_client.py (hunk 3 above)
Comment (rogthefrog) [reply plugins/vertexai/modelgauge/suts/vertexai_client.py:66]: You don't need the gcloud CLI itself, but you do need the application default credentials. I'll add that to the documentation.

**Review by bkorycki — APPROVED** [review]
I'm not sure that the authentication in the vertex AI plugin works the way we would want. But I don't want to hold up merging this PR since we won't actually be using that plugin for any upcoming runs, so we should just revisit that later on.

**Pull request merged**